### PR TITLE
New media events

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Adding new MediaEvents to follow more closely the model we have in Ophan and model we can use in GA.

Nothing is using these events at the moment - but the plan is to get them into GA, and then potentially port them to Ophan.
## What is the value of this and can you measure success?

Things become simpler as time goes on.
## Does this affect other platforms - Amp, Apps, etc?

No
## Screenshots

🎅 
## Request for comment

@akash1810 @gidsg

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -166,6 +166,7 @@ define([
             withPreroll,
             blockVideoAds;
 
+
```
Comment (gidsg) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: I do not feel this whitespace adds a huge amount of value.

Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Agreed. It has been terminated.

File: static/src/javascripts/projects/common/modules/video/events.js [hunk 2]
```diff
@@ -36,6 +40,93 @@ define([
             'content:end'
         ];
 
+
+    /**
+     *
+     * @param mediaId {string}
+     * @param mediaType {string} audio|video
+     * @param eventType {string} e.g. firstplay, firstend
+     * @param preroll {boolean}
```
Comment (akash1810) [inline static/src/javascripts/projects/common/modules/video/events.js]: minor: would `isPreroll` make the data-type more obvious?

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 3]
```diff
@@ -200,6 +200,11 @@ define([
                 }
             }
         }));
+        events.addContentEvents(player, mediaId, mediaType);
+        events.addPrerollEvents(player, mediaId, mediaType);
+        if (window.location.hash === '#debugVideoEvents') {
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: should this be an item in `frontend.properties` instead?

Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: would require a server restart for changes to be picked up, but would behave more like the `useHashedBundles` flag?

**Comment by akash1810** [comment]
couple of minor comments, else 👍

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 3 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Not 100% sure what you're talking about - here we're just looking at the hash?

**Pull request merged**